client/state/selectors/index.js

```diff
@@ -60,6 +61,7 @@ export isDeactivatingJetpackModule from './is-deactivating-jetpack-module';
 export isDomainOnlySite from './is-domain-only-site';
 export isFetchingJetpackModules from './is-fetching-jetpack-modules';
 export isJetpackModuleActive from './is-jetpack-module-active';
+export isJetpackModuleUnavailableInDevelopmentMode from './is-jetpack-module-unavailable-in-development-mode';
```

Comment:
First, this is copying from the functionality in https://github.com/Automattic/jetpack/pull/6180, which uses the same pattern. Second, in most cases that we'll need to implement this selector, we'll need to check against the negative case, for example:

* We'll display a nudge that the site is in development mode only if the module is unavailable in dev mode
* We'll disable the form controls within a module only if the module is unavailable in dev mode.

That said, if we make it `isJetpackModuleAvailableInDevelopmentMode`, we'll have to use it `! isJetpackModuleAvailableInDevelopmentMode` everywhere, so it makes sense for it to check against the negative case.